docs/CANDIDATE_TECHNOLOGIES.md

```diff
@@ -0,0 +1,159 @@
+
+#### Advantages
+
+- Based on JavaScript.
+- OS-level execution of actions.
+- Simulate keyboard as well as mouse events.
+- Requires no separate execution of scripts by the user.
+- Low cost (under $50).
+
+#### Disadvantages
+
+- Requires a lot of technical setup and integration by the user.
+- Cannot use Infusion due to memory limitations.
+- Might require additional hardware (buttons, switches, pins, etc.) to work.
+- Can be easily damaged or destroyed (bricked) by the user.
+- Requires knowledge from the user to debug or fix, if damaged (bricked).
+- Less flexible in terms of reconfiguration.    <!-- Swap hardware, buttons, etc. -->
+
+Although this is a low-cost solution, it seems inconvenient to me, both as a developer and a user. This would require too much of technical setup and integration and a decent knowledge of IoT Boards. Users might feel overwhelmed.
+
+### Secondary Extensions / Add-ons
+
+There are some extensions or add-ons that can inject our code (scripts) into the webpages. Some of the popular ones are [Tampermonkey](https://chrome.google.com/webstore/detail/tampermonkey/dhdgffkkebhmkfjojejmpbldmpobfkfo?hl=en) (For Safari and Chromium-based browsers) and [Greasemonkey](https://addons.mozilla.org/en-US/firefox/addon/greasemonkey/) (for Firefox).
+
+#### Advantages
+
+- Based on JavaScript and can support Infusion.
+- Can run on both Desktop and Mobile (supported for a few browsers only).
+
+#### Disadvantages
+
+- Requires setting-up and loading the scripts separately by the user.
+- Limitations in regards to a few navigation features or effects.
+- More dependent on technology.
+- Sometimes load potentially unwanted programs (PUPs) and malware in the background.
+- Contains some potential bugs that can hamper the user experience.
+- Less flexible in terms of reconfiguration.
+
+### Bookmarklets
+
+A bookmarklet is a bookmark tab that injects script into the current webpage when pressed. The bookmark tab must be linked to the script that we want to insert, and the script must be hosted on the web.
+
+#### Advantages
+
+- Based on JavaScript and can support Infusion.
+- Requires no additional software/hardware to install and set-up.
+- Can run on both Desktop and Mobile.
+- Supported by all the browsers.
+
+#### Disadvantages
+
+- Requires the user to set-up the bookmarklet manually.
+- Needs to be executed manually every time the webpage is updated, reloaded, or a new tab or browser window is opened.
+- Limitations in regards to a few navigation features or effects.
+- Less flexible in terms of reconfiguration.
+
+## Conclusion
+
+I propose to use a Chrome Extension or an Electron.js App for the project. Espruino Pico is also a potential candidate, but the former two seem to be a better option.
```

Comment:
In my own work, when I have narrowed things down to a smaller set of candidates, I often try them out a bit to help with the final selection.  For whichever technology you select, the research you've done (trying libraries, reading docs) gives you a head start on learning, and it's usually not a bad idea to at least be aware of patterns used in alternative technologies.